src/compute/src/rpc/service/stream_service.rs

```diff
@@ -177,15 +177,27 @@ impl StreamService for StreamServiceImpl {
         request: Request<BarrierCompleteRequest>,
     ) -> Result<Response<BarrierCompleteResponse>, Status> {
         let req = request.into_inner();
-        let (collect_result, checkpoint) = self
+        let (
+            CollectResult {
+                create_mview_progress,
+                is_first_barrier,
+            },
+            checkpoint,
+        ) = self
             .mgr
             .collect_barrier(req.prev_epoch)
             .instrument_await(format!("collect_barrier (epoch {})", req.prev_epoch))
             .await
             .inspect_err(|err| tracing::error!("failed to collect barrier: {}", err))?;
         // Must finish syncing data written in the epoch before respond back to ensure persistence
         // of the state.
-        let synced_sstables = if checkpoint {
+        let synced_sstables = if is_first_barrier {
+            tracing::info!(
```

Comment:
Shall we add a check here to ensure that when `is_first_barrier`, the `req.prev_epoch` equals to the epoch of current hummock version? 

We can get the hummock version by doing a match on the `self.env.state_store` and on case `Hummock` we get the hummock version by `hummock.get_pinned_version()`.

Reply:
Good point.

Reply:
I find that even though there's no actor in the cluster, we still trigger periodical barriers and update the committed epoch in `HummockManager`. So the first barrier received by the compute node is typically not the very first one in the cluster, then the assertion cannot hold due to asynchronous notification of `PinnedVersion`. 🤔

Reply:
Do you mean that when there is no actor in the cluster, for the periodically triggered barrier, we will not send `collect_barrier` to CN and directly commit an epoch on meta node?

Reply:
Yes.

Reply:
Then can we explicitly call a try_wait_epoch(req.prev_epoch) for the first barrier to ensure that at the beginning the epoch on CN are aligned with the meta node?

Reply:
Oh sorry, I thought the `is_first_barrier` is set by the mete node, but it seems that it is set by the CN.

Reply:
> Then can we explicitly call a try_wait_epoch(req.prev_epoch) for the first barrier to ensure that at the beginning the epoch on CN are aligned with the meta node?

Yes, we can, but it seems not necessary. 🤔

For correctness: since we have already handled `wait_epoch_commit` for both scaling and recovery, the actors in the remaining cases always start fresh. Therefore, there is no need to wait for any previous version to be synced.

For assertion: `wait`ing then asserting epochs match seems to be trivial, since it's not possible to see a future version at that moment. As long as `wait` returns, we always get the version with `prev_epoch`.

Reply:
> I thought the `is_first_barrier` is set by the mete node

Yes, this is also a possible solution, maybe I can have a try since it seems a little bit clearer.

Reply:
> I thought the `is_first_barrier` is set by the mete node

Done and the assertion is added back.